sbi/diagnostics/lc2st.py

```diff
@@ -0,0 +1,730 @@
+        """
+
+        # prepare data
+
+        if self.z_score:
+            P = (P - self.P_mean) / self.P_std
+            Q = (Q - self.P_mean) / self.P_std
+            x_P = (x_P - self.x_P_mean) / self.x_P_std
+            x_Q = (x_Q - self.x_P_mean) / self.x_P_std
+
+        # initialize classifier
+        clf = self.clf_class(**self.clf_kwargs or {})
+
+        # cross-validation
+        if self.n_folds > 1:
+            trained_clfs = []
+            kf = KFold(n_splits=self.n_folds, shuffle=True, random_state=self.seed)
+            cv_splits = kf.split(self.P)
+            for train_idx, _ in tqdm(
+                cv_splits, desc="Cross-validation", disable=verbosity < 1
+            ):
+                # get train split
+                P_train, Q_train = P[train_idx], Q[train_idx]
+                x_P_train, x_Q_train = x_P[train_idx], x_Q[train_idx]
+
+                # train classifier
+                clf_n = train_lc2st(P_train, Q_train, x_P_train, x_Q_train, clf)
+
+                trained_clfs.append(clf_n)
+        else:
+            # train single classifier
+            clf = train_lc2st(P, Q, x_P, x_Q, clf)
+            trained_clfs = [clf]
+
+        return trained_clfs
+
+    def _scores(
+        self,
+        P_eval: Tensor,
+        x_eval: Tensor,
+        trained_clfs: List[Any],
+        return_probas: bool = False,
+    ) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+        """Compute the L-C2ST scores given the trained classifiers.
+
+        Args:
+            P_eval: Samples from P conditioned on the observation `x_eval`,
+                of shape (sample_size, dim).
+            x_eval: Observation, of shape (,dim_x).
+            trained_clfs: Trained classifiers.
+            return_probas: Whether to return the predicted probabilities of being in P,
+                defaults to False.
+
+        Returns: one of
+            scores: L-C2ST scores at `x_eval`.
+            (probas, scores): Predicted probabilities and L-C2ST scores at `x_eval`.
+        """
+        # prepare data
+        if self.z_score:
+            P_eval = (P_eval - self.P_mean) / self.P_std
+            x_eval = (x_eval - self.x_P_mean) / self.x_P_std
+
+        probas, scores = [], []
+
+        # evaluate classifiers
+        for clf in trained_clfs:
+            proba, score = eval_lc2st(P_eval, x_eval, clf, return_proba=True)
+            probas.append(proba)
+            scores.append(score)
+        probas, scores = np.array(probas), np.array(scores)
+
+        if return_probas:
+            return probas, scores
+        else:
+            return scores
+
+    def train_data(self) -> None:
+        """Train the classifiers on the observed data.
+        Saves the trained classifiers.
+        """
+        trained_clfs = self._train(self.P, self.Q, self.x_P, self.x_Q)
+        self.trained_clfs = trained_clfs
+
+    def scores_data(
+        self, P_eval: Tensor, x_eval: Tensor, return_probas: bool = False
```

Comment:
Some renaming suggestions to conform with our conventions: 
```suggestion
        self, theta_o: Tensor, x_o: Tensor, return_probs: bool = False
```

or use `posterior_samples` instead of `theta_o`.